### PR TITLE
Handle generic blocks transformers and add "smart fields"  concept

### PR DESCRIPTION
**Pull request by cekk**
Transformers registered with **None** type can be run for every block.

I registered two new transformers (one serializer and one deserializer) to handle resolveuid for generic blocks with a "standard" field (**url** or **href**).

The other thing that i added is that if a block has a "searchableText" field, we add its value to SearchableText index of the context.

@tisto @sneridagh @tiberiuichim do we need to discuss more on this?

### CODE REVIEW

**Comment by cekk** [comment]
@jenkins-plone-org please run jobs

**Comment by cekk** [comment]
@tisto i renamed this pr.
Can you give it a review? I'd like to use these features ;)

**Comment by tisto** [comment]
@cekk Sorry for not looking into this within a proper timeframe! I will put it on my todo list!

**Review by tiberiuichim — APPROVED** [review]

**Comment by tiberiuichim** [comment]
@jenkins-plone-org please run jobs

**Review by tiberiuichim — APPROVED** [review]

File: src/plone/restapi/deserializer/blocks.py [hunk 1]
```diff
@@ -60,6 +74,29 @@ def __call__(self, value):
         return value
 
 
+@adapter(IBlocks, IBrowserRequest)
+@implementer(IBlockFieldDeserializationTransformer)
+class ResolveUIDDeserializer(object):
+    """
+    This is a general handler. It will be loaded for all blocks
```
Comment (tisto) [inline src/plone/restapi/deserializer/blocks.py:81]: general -> generic

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

**Pull request merged**